src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/host-form/host-form.component.ts

```diff
@@ -77,30 +89,65 @@ export class HostFormComponent extends CdForm implements OnInit {
     });
   }
 
+  // expand hostnames in case hostname is a pattern
+  checkHostNamePattern(hostname: string) {
+    const braceExpansion = require('brace-expansion');
+    if (hostname.includes('[') && hostname.includes(']') && !hostname.match(/(?![^(]*\)),/g)) {
+      const hostnameRange = hostname
```

Comment:
I find this block perhaps too complicate.

For example, you have already defined a function to detect whether it's a pattern or no not, right? `checkHostNameValue()`... However you cannot use that because that function do too maany things (or at least more than one). That's my suggestion to use simple functions with simple interfaces, and encapsulate them in more complex functions.

This should start like:
```ts

if (isHostnamePattern()) { // you already have that checker
  // then to pattern stuff
} else {
...
}
```